### ISSUE TITLE
G-Suite Email Setup (IMAP/SMTP)

### ISSUE THREAD

**Issue by ovrdrvn** [osTicket/osTicket#4769]
I have the proper port (993, 465), credentials etc and if I use imap.gmail.com and ssl://smtp.gmail.com it spins for awhile and I get a 405 error

login name is full email address as instructed

**Comment by ovrdrvn** [comment]
Just realized...two factor authentication the issue? not getting a text or such so wasn't aware if it's even hitting the server

**Comment by JediKev** [comment]
@ovrdrvn

Yes. You need to read google’s guide:
https://support.google.com/accounts/answer/185833?hl=en

Cheers.

**Comment by ovrdrvn** [comment]
@JediKev Got an app password and put it in. Still spinning for a minute plus and I get a 405. Normally, you put smtp.gmail.com and there is some box but i'm doing it as above. Imap should be straight forward. Any other ideas?

**Comment by JediKev** [comment]
@ovrdrvn

Look at you server’s maillogs for any errors.

Cheers.

**Comment by ovrdrvn** [comment]
@JediKev My server host is killing me (it is a shared environment but they claim to support osTicket) I am waiting on them to provide logs. I've tried incoming and outgoing separately and asked them to check ports and SSL (not sure what else could be blocked) btw, it's Bluhost in case you are familiar with them

**Comment by ovrdrvn** [comment]
@JediKev Do I need to set up Remote Piping to use Gmail to update our tickets? I'm still convinced they are blocking ports as I can't telnet, but just read this piece about piping.

**Comment by TechCare-Aus** [comment]
@ovrdrvn I use G Suite and WHM/cPanel. By default the firewall blocks port 465 outgoing (go figure??). Seeing that I host my own VPS I can adjust that myself and then it worked fine. Although I'm not using 2FA.

What is the hosting software that you are using?

**Comment by TechCare-Aus** [comment]
@ovrdrvn set up a cron job. This is what I use in cPanel Cron:

I've set it to run at 1 min intervals.

```/usr/bin/php /your/location/for/osTicket/api/cron.php > /your/location/for/logs/cron_osticket.log 2>&1```

```/usr/bin/php``` = calls php to run the following command... ```/your/location/for/osTicket/api/cron.php``` 

```>``` = replace output file with this output, you can also use ```>>``` which will append the output of running the cron.php to your log file (handy for dealing with multiple errors over time) but increases the size of the log file each time

```2>&1``` = [go here ](https://unix.stackexchange.com/questions/99263/what-does-21-in-this-command-mean)

Also make sure that your settings in Admin Panel > Emails > Settings have the box "**Email Fetching**" ticked!

**Comment by ovrdrvn** [comment]
@TechCare-Aus Thanks! I'm still stuck not being able to save the email IMAP settings. The host is now trying to help but I can't add another admin user and get the error: Unable to add this agent. Correct any errors below and try again. Yet there are no issues listed below. I'm wondering if something greater is wrong with the install even though the rest seems to function well.

**Comment by JediKev** [comment]
@ovrdrvn

I’m not able to reproduce this issue so please provide related errors via error logs. Also, try to do a search on GitHub for the same issue as there might already be one open with a fix. If there isn’t one then please open a separate issue to avoid multiple, separate issues in one thread. 

Cheers.

**Comment by TechCare-Aus** [comment]
@ovrdrvn as @JediKev has informed you please provide the logs. I know in my cPanel (which it appears that bluehost use as well) you can open the File Manager app in your hosting login. Then navigate to /home/*YOUR*USERNAME*/logs/ in here you should fine your php logs where it should tell you what the error message means in more detail. 

But your best bet is to use the 24/7 Bluehost chat and ask them about it. 

Also what version of osTicket are you running? Go to Admin Panel > Dashboard > Information and post back here everything (minus any identifying information).

**Comment by ovrdrvn** [comment]
@TechCare-Aus @JediKev Thank you both for your guidance. It turned out Bluehost staff had no idea that their shared hosting blocked ports. We were moved to a VPS and now the ports are open. Before closing this email related ticket (as I do have another question so will start a different thread) once setting up the IMAP, it seems the system was pulling in all the old email in the account and thus creating bounced tickets. Is there some means of preventing this?

**Comment by TechCare-Aus** [comment]
@ovrdrvn if i understand you correctly the easiest way is to have a clean INBOX and not confuse things is to set the option to move imported emails into another folder. In addition tHis mailbox shouldmt be accessed at all by anything except osTicket unless to diagnose an issue.

So for me I have: 
INBOX - where all new emails or replies go
INBOX\RETRIEVED - where osTicket places emails after it has "read" and imported them. This folder to my knowledge doesn't need to exist just configure it via osTicket > Admin Panel > Email > Email > Select Email Address > Fetched Emails > Move to folder: RETRIEVED (or whatever you like)

**Comment by ovrdrvn** [comment]
@TechCare-Aus So we have THOUSANDS of old emails that I'd prefer not get imported as the work is long done. There is no way to avoid this? It's IMAP so the emails are on the server of course. We would like to turn on the new helpdesk and work with emails moving forward. I gather turning on the email settings starts retrieving all the old emails. Also, is it safe to assume that promotional emails, junk and things not client related will be ignored by osTicket if the email addresses aren't in its database?

**Comment by ntozier** [comment]
You can delete them or move them to an answered (or whatever you want to call it) folder.

**Comment by ovrdrvn** [comment]
@ntozier There is approx 10 years worth of email that is a substantial amount...if I have to let them go into the database, I guess I better check how much space we have first.  

Got it on the fail part and we are having the agents register all users (can't find a bulk import feature but it would help).

**Comment by ntozier** [comment]
Agent Panel -> Users -> Import

![2019-03-05_1159](https://user-images.githubusercontent.com/1315768/53822699-3fa7e200-3f3e-11e9-9782-772d9e09c6cd.png)

**Comment by ovrdrvn** [comment]
Thanks and moving on to hopefully getting this fully functional!
